aws/resource_aws_lb_listener.go

```diff
@@ -147,7 +154,38 @@ func resourceAwsLbListenerCreate(d *schema.ResourceData, meta interface{}) error
 		return errors.New("Error creating LB Listener: no listeners returned in response")
 	}
 
-	d.SetId(*resp.Listeners[0].ListenerArn)
+	larn := *resp.Listeners[0].ListenerArn
+
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		if carns, ok := d.GetOk("additional_certificate_arns"); ok {
+			for _, carn := range expandStringList(carns.(*schema.Set).List()) {
+				_, err := elbconn.AddListenerCertificates(&elbv2.AddListenerCertificatesInput{
+					Certificates: []*elbv2.Certificate{&elbv2.Certificate{
+						CertificateArn: carn,
+					}},
+					ListenerArn: aws.String(larn),
+				})
+				if awsErr, ok := err.(awserr.Error); ok {
+					if awsErr.Code() == "TooManyCertificates" || awsErr.Code() == "CertificateNotFound" {
```

Comment:
What is the reason for retrying on `TooManyCertificates` here? 🤔

Reply:
Only as I essentially copied `CertificateNotFound`. Neither makes sense to retry on IMO.